### DISCUSSION TITLE
Narrowing Prisma return types with filter conditions

### DISCUSSION THREAD

**Discussion by kylegl** [prisma/prisma#23080] [Q&A] [answered]
### Question

Problem: Prisma  query return type doesn't reflect filter conditions.

```
model User {
	 name  String?
}

const users = await prisma.users.findMany({
	where: { name: { not: null } },
})

//   ^? type users = { name: string | null }[]
```

Id expect the result type to be `{ name: string }[]`.

Is this a limitation of prisma or is there a way to return the proper type in this situation?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @kylegl 👋 

Thank you for raising this question.

We do have a similar [issue](https://github.com/prisma/prisma/issues/9508) that raises the question you asked. At the moment, the return type of a query does not currently change based on the filters used in the query. So even if you filter out null values, the return type will still include null as a possible type.

Can you please try the workaround mentioned [here](https://github.com/prisma/prisma/issues/9508#:~:text=Alternatives-,A%20work%20around%20is,-import%20%7B%20User%20%7D%20from)?

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
